src/components/NavBar/NavBar.jsx

```diff
@@ -9,48 +9,52 @@ function NavBar({ token, theme, onToggleTheme }) {
   return (
     <div className={classes.navBar} data-theme={theme}>
       <div className={classes.navBar_container}>
-        <div>
+        <div className={classes.navBar_logo_wrapper}>
           <NavLink to="/" className={classes.navBar_logo}>
             <img src={weatherLogo} alt="weather logo" className={classes.weather_logo} />
             FakeForeca
           </NavLink>
         </div>
 
-        <div className={classes.search_menu}>
-          <SearchInput token={token} theme={theme} />
+        <div className={classes.search_navBar_wrapper}>
+          <div className={classes.search_menu}>
+            <SearchInput token={token} theme={theme} />
+          </div>
 
-          <ul className={classes.navBar_menu}>
-            <li className={classes.navBar_item}>
-              <NavLink
-                exact
-                to="/"
-                className={classes.navBar_link}
-                activeClassName={classes.navBar_active_link}
-              >
-                Home
-              </NavLink>
-            </li>
-            <li className={classes.navBar_item}>
-              <NavLink
-                to="/info"
-                className={classes.navBar_link}
-                activeClassName={classes.navBar_active_link}
-              >
-                Info
-              </NavLink>
-            </li>
-            <li className={classes.navBar_item}>
-              <NavLink
-                to="/feedback"
-                className={classes.navBar_link}
-                activeClassName={classes.navBar_active_link}
-              >
-                Feedback
-              </NavLink>
-            </li>
-          </ul>
+          <div className={classes.navBar_menu_wrapper}>
+            <ul className={classes.navBar_menu}>
+              <li className={classes.navBar_item}>
+                <NavLink
+                  exact
+                  to="/"
+                  className={classes.navBar_link}
+                  activeClassName={classes.navBar_active_link}
+                >
+                  Home
+                </NavLink>
+              </li>
+              <li className={classes.navBar_item}>
+                <NavLink
+                  to="/info"
+                  className={classes.navBar_link}
+                  activeClassName={classes.navBar_active_link}
+                >
+                  Info
+                </NavLink>
+              </li>
+              <li className={classes.navBar_item}>
+                <NavLink
+                  to="/feedback"
+                  className={classes.navBar_link}
+                  activeClassName={classes.navBar_active_link}
+                >
+                  Feedback
+                </NavLink>
+              </li>
+            </ul>
 
-          <ThemeSwitcher theme={theme} onToggleTheme={onToggleTheme} />
+            <ThemeSwitcher theme={theme} onToggleTheme={onToggleTheme} />
```

Comment:
please use context for accessing and updating theme, so every component could get "theme" value independently from their parents props